File: 2023/langs/python/1/solution1.py
import os

with open(os.path.join(os.getcwd(), "..", "..", "..", "inputs", "1.txt"), "r") as file:
    contents = file.read()

calibrationData = map(lambda line: list(line), contents.split("\n"))
numbers = list("1234567890")

total = 0

for value in calibrationData:
    digits = ""

    for i in range(len(value)):
        digit = value[i]
        if digit not in numbers:
            continue

        digits += digit
        break

    for i in reversed(range(len(value))):
        digit = value[i]
        if digit not in numbers:
            continue

        digits += digit
        break

    total += int(digits)

print(total)

File: 2023/langs/js/1/solution1.mjs
import { readFile } from "node:fs/promises";
import { join } from "node:path";
import { cwd } from "node:process";

const filePath = join(cwd(), "..", "..", "..", "inputs", "1.txt");
const contents = await readFile(filePath, "utf-8");
const calibrationData = contents.split("\n");
const numbers = "1234567890".split("");

let total = 0;

for (const value of calibrationData) {
  let digits = "";
  for (let i = 0; i < value.length; i++) {
    const digit = value[i];
    if (!numbers.includes(digit)) continue;

    digits += digit;
    break;
  }

  for (let i = value.length - 1; i >= 0; i--) {
    const digit = value[i];
    if (!numbers.includes(digit)) continue;

    digits += digit;
    break;
  }

  total += +digits;
}

console.log(total);
